unittests/AST/ASTImporterTest.cpp

```diff
@@ -3173,6 +3173,105 @@ TEST_P(
                     .match(ToTU, classTemplateSpecializationDecl()));
 }
 
+class ImportImplicitMethods : public ASTImporterTestBase {
+public:
+  static constexpr auto DefaultCode = R"(
+      struct A { int x; };
+      A&& fa() { return static_cast<A&&>(A()); }
+      void f() {
+        A a;
+        A a1(a);
+        A a2(fa());
+        a = a1;
+        a = fa();
+        a.~A();
+      })";
+
+  template<typename MatcherType>
+  void test(const MatcherType &MethodMatcher,
+      const char *Code = DefaultCode, bool CheckNoImport = false) {
+
+    Decl *ToTU = getToTuDecl(Code, Lang_CXX11);
+    auto *ToClass = FirstDeclMatcher<CXXRecordDecl>().match(
+        ToTU, cxxRecordDecl());
+
+    ASSERT_EQ(DeclCounter<CXXMethodDecl>().match(ToClass, MethodMatcher), 1);
+
+    {
```

Comment:
Do we need the blocks? Is there a name collision, or you just want to express that `Method` is not used in the final expectations?

Reply:
This is a kind of sub-operation with local variables for it. (Other tests have same constructs, probably because name conflicts). The `Method` is anyway not needed later, it points to a removed thing that should not be used any more.